### PR TITLE
Spark features should not drop spark configs if no features are found

### PR DESCRIPTION
**Pull request by catalinii**
# TL;DR
https://github.com/flyteorg/flyte/issues/696

if the feature is not matched, the sparkConfig is appended with the actual config.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ x] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/696

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by catalinii** [comment]
I do not have permissions to push branches to the this repo it seems.

File: go/tasks/plugins/k8s/spark/spark.go [hunk 1]
```diff
@@ -217,23 +217,29 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
...
 	featureName := matches[0][len(matches[0])-1]
+	found := false
 	// Use the first matching feature in-case of duplicates.
 	for _, feature := range GetSparkConfig().Features {
 		if feature.Name == featureName {
+			found = true
 			for k, v := range feature.SparkConfig {
 				sparkConfig[k] = v
 			}
 			break
```
Comment (akhurana001) [inline go/tasks/plugins/k8s/spark/spark.go:238]: nit: Maybe just return here and no need for `found`

Comment (catalinii) [reply go/tasks/plugins/k8s/spark/spark.go]: Addrrsses

**Comment by kumare3** [comment]
@akhurana001 is it a +1?

**Review by akhurana001 — APPROVED** [review]

**Comment by akhurana001** [comment]
LGTM

**Pull request merged**